ts/screens/authentication/IdpLoginScreen.tsx

```diff
@@ -145,11 +151,16 @@ class IdpLoginScreen extends React.Component<Props, State> {
     this.setState({
       requestState: event.loading ? pot.noneLoading : pot.some(true)
     });
+  };
 
-    onNavigationStateChange(
+  private handleShouldStartLoading = (event: NavState): boolean => {
+    // onNavigationStateChange returns true if url matches login pattern and it contains token
+    // if the url means login success, we don't care about loading that url
+    const stopNavigation = onNavigationStateChange(
```

Comment:
better naming, related to the actual meaning of the value returned by `onNavigationStateChange`:

```suggestion
    const isLoginUrlWithToken = onNavigationStateChange(
```